### PR TITLE
PPF-452 Make key visibility more clear for admins

### PR DESCRIPTION
**Pull request by JonasVHG**
### Changed

- `app/Domain/Integrations/Integration`: Add `isKeyVisibleForEnvironment()`
- `app/Nova/Resources/Auth0Client`: Add column with `Visible for integrator` in adminPanel.
- `app/Nova/Resources/KeycloakClient`: Add column with `Visible for integrator` in adminPanel.
- `app/Nova/Resources/UiTiDv1`: Add column with `Visible for integrator` in adminPanel.

### Fixed

- `KeyVisibility` is now more clear for Admins.

---
Ticket: https://jira.publiq.be/browse/PPF-452

### CODE REVIEW

File: app/Nova/Resources/Auth0Client.php [hunk 1]
```diff
@@ -67,6 +71,19 @@ public function fields(NovaRequest $request): array
                     Auth0Tenant::Testing->value => Auth0Tenant::Testing->name,
                     Auth0Tenant::Production->value => Auth0Tenant::Production->name,
                 ]),
+            Text::make('Visible for integrator', static function (Auth0ClientModel $model) {
+                $auth0Client = $model->toDomain();
+                /** @var Integration $integration */
+                $integration = App::get(IntegrationRepository::class)->getById($auth0Client->integrationId);
+                $isVisible = $auth0Client->tenant !== Auth0Tenant::Acceptance &&
```
Comment (LucWollants) [inline app/Nova/Resources/Auth0Client.php:78]: I think you could add an extra method `Integration::isKeyVisibleForEnvironment`. It requires only a conversion from a specific environment to the environment inside the Integration namespace. But all those enums have the same value.

Comment (JonasVHG) [reply app/Nova/Resources/Auth0Client.php:78]: Wouldn't you also need to take the `keyVisibility` into account?

Comment (JonasVHG) [reply app/Nova/Resources/Auth0Client.php:78]: But maybe I can get the visibilty from environment.

Comment (LucWollants) [reply app/Nova/Resources/Auth0Client.php:78]: But the visibility is taken from the integration I think, see `$integration->getKeyVisibility()`

Comment (JonasVHG) [reply app/Nova/Resources/Auth0Client.php]: I have implemented it, but I do not think it's ideal.

File: app/Domain/Integrations/Integration.php [hunk 2]
```diff
@@ -230,6 +232,12 @@ public function urlsForTypeAndEnvironment(IntegrationUrlType $type, Environment
         ));
     }
 
+    public function isKeyVisibleForEnvironment(UiTiDv1Environment|Auth0Tenant|Environment $environment): bool
```
Comment (LucWollants) [inline app/Domain/Integrations/Integration.php:235]: I think you can even do this

```
    public function isKeyVisibleForEnvironment(UiTiDv1Environment|Auth0Tenant|Environment $environment): bool
    {
        $keyVisibility = KeyVisibility::v1;
        if ($environment instanceof UiTiDv1Environment) {
            $keyVisibility = KeyVisibility::v2;
        }

        return $environment->value !== 'acc' &&
            $this->status !== IntegrationStatus::Deleted;
            $this->status !== IntegrationStatus::Deleted &&
            $this->getKeyVisibility() !== $keyVisibility;
    }
```

Comment (LucWollants) [reply app/Domain/Integrations/Integration.php:235]: Or even work with a ternary operator.

**Review by LucWollants — COMMENTED** [review]
Looks good, thanks for the extra refactor to a reusable method.

**Review by LucWollants — APPROVED** [review]

**Pull request merged**